txpool/pool.go

```diff
@@ -2187,19 +2193,28 @@ func (mt *metaTx) better(than *metaTx, pendingBaseFee uint64) bool {
 
 	switch mt.currentSubPool {
 	case PendingSubPool:
-		var effectiveTip, thanEffectiveTip uint64
-		if pendingBaseFee <= mt.minFeeCap {
-			effectiveTip = cmp.Min(mt.minFeeCap-pendingBaseFee, mt.minTip)
+		var effectiveTip, thanEffectiveTip uint256.Int
+		if mt.minFeeCap.Cmp(&pendingBaseFee) >= 0 {
+			if mt.minFeeCap.Cmp(uint256.NewInt(mt.minTip)) <= 0 {
```

Comment:
You need a temporary variable to calculate the difference `mt.minFeeCap-pendingBaseFee`